test/integration/logbucket-project/logbucket_project_test.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logbucket_project
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/gcloud"
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/tft"
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/utils"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestLogBucketProjectModule(t *testing.T) {
+
+	const logApiFdqm = "logging.googleapis.com"
+	const RETENTION_DAYS = 20
+	const ROLE_BUCKET_WRITER string = "roles/logging.bucketWriter"
+
+	bpt := tft.NewTFBlueprintTest(t,
+		tft.WithTFDir("../../../examples/logbucket/project"),
+	)
+	bpt.DefineVerify(func(assert *assert.Assertions) {
+		bpt.DefaultVerify(assert)
+
+		//*******************************
+		// Bucket on other project test *
+		//*******************************
+		projId := bpt.GetStringOutput("log_bucket_project")
+		bktName := bpt.GetStringOutput("log_bucket_name")
+		bktDestProjId := bpt.GetStringOutput("log_bkt_same_proj")
+		sinkProjId := bpt.GetStringOutput("log_sink_project_id")
+		sinkDest := bpt.GetStringOutput("log_sink_destination_uri")
+		sinkName := bpt.GetStringOutput("log_sink_resource_name")
+		sinkWriterIdentity := bpt.GetStringOutput("log_sink_writer_identity")
+
+		logBucketDetails := gcloud.Runf(t, fmt.Sprintf("logging buckets describe %s --location=%s --project=%s", bktName, "global", projId))
+
+		// assert log bucket name, retention days & location
+		assert.Equal(sinkDest[len(logApiFdqm)+1:], logBucketDetails.Get("name").String(), "log bucket name should match")
```

Comment:
what is this code doing?
```
sinkDest[len(logApiFdqm)+1:]
```

Reply:
I think it would be more clear if this was
```
assert.Contains(sinkDest, logBucketDetails.Get("name").String() ...
```

Reply:
Changed assert bucket name to remove substring and compare to a built string fully qualified name